django/chat/models.py

```diff
@@ -14,14 +14,22 @@
     QA_PRE_INSTRUCTIONS,
     QA_PROMPT_TEMPLATE,
     QA_SYSTEM_PROMPT,
+    current_time_prompt,
 )
 from librarian.models import DataSource, Library, SavedFile
 from otto.models import SecurityLabel
 from otto.utils.common import display_cad_cost, set_costs
 
 logger = get_logger(__name__)
 
-DEFAULT_MODE = "qa"
+DEFAULT_MODE = "chat"
+
+
+def create_chat_data_source(user):
+    return DataSource.objects.create(
+        name=f"Chat {timezone.now().strftime('%Y-%m-%d %H:%M:%S')}",
+        library=user.personal_library,
+    )
```

Comment:
Refactoring as suggested by Jason to reverse relationship between Chat and DataSource.
Chat now has a data_source property which is populated with this helper when Chat is created (or, for existing chats, when chat is accessed).